torch/_inductor/scheduler.py

```diff
@@ -534,7 +533,7 @@ def get_first_name(self) -> str:
 
     @cache_on_self
     def get_names(self) -> Set[str]:
-        return functools.reduce(set.union, [x.get_names() for x in self.snodes])
+        return set.union(*[x.get_names() for x in self.snodes])
```

Comment:
Even better
```suggestion
        return set.union(x.get_names() for x in self.snodes)
```
same in the other two instances below.

Reply:
ok, so `set.union` doesn't accept generators. You could still do `*(x.get...)`, but yeah.